mutex.go

```diff
@@ -0,0 +1,97 @@
+package stcql
+
+import (
+	"fmt"
+	"time"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/gocql/gocql"
+)
+
+// Mutex represents a mutex that is acquired by inserting a row
+// into a Cassandra table using a lightweight transaction.
+type Mutex struct {
+	Session    *gocql.Session
+	Keyspace   string
+	Table      string
+	LockName   string
+	LockHolder string
+	// RetryTime is the duration between attempts to acquire the mutex
+	RetryTime time.Duration
+	// Timeout is the duration after which to give up acquiring the mutex
+	Timeout time.Duration
+}
+
+// Lock attempts to acquire mutex m. If the lock is already in use, the calling
+// gorouting blocks until the mutex is available. If the duration m.Timeout passes
+// while Lock is executing, a runtime panic will occur.
+func (m *Mutex) Lock() {
+	if m.tryLock() {
+		return
+	}
+	retryTicker := time.Tick(m.RetryTime)
+	var timeoutTicker <-chan time.Time
+	if m.Timeout > 0 {
+		timeoutTicker = time.Tick(m.Timeout)
+	} else {
+		// if m.Timeout isn't a positive value, just make an unconnected channel to poll
+		// (effectively making the timeout infinite)
+		timeoutTicker = make(chan time.Time)
+	}
+	for {
+		select {
+		case <-retryTicker:
+			if m.tryLock() {
+				return
+			}
+		case <-timeoutTicker:
+			panic(fmt.Sprintf("mutex lock hit timeout of %v", m.Timeout))
```

Comment:
I don't love panicking either but I'm not totally sure of the best practice in a library. If I `panic` here rather than `log.Fatal`, the caller has the option of using `recover` to catch it. I was sort of using `sync.Mutex` as a guide -- for example, [`sync.Mutex.Unlock()`](https://golang.org/pkg/sync/#Mutex.Unlock) will panic if you try to unlock an unlocked Mutex.